documentation/docs/guides/isolated-development-environments.md

```diff
@@ -0,0 +1,223 @@
+---
+title: Isolated Development Environments
+sidebar_position: 25
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+This guide shows you how to set up isolated development environments using the **[Container Use MCP](https://github.com/dagger/container-use)** with Goose. With this setup, your development work will be isolated to both git branches and containers, allowing you to experiment freely without affecting your main system state.
+Note that the container-use MCP is very new and emerging, but offers powerful ways to do isolated development which are very agent friendly (build on tools like Docker, copy on write filesystems and more)
+
+## Overview
+
+The **[Container Use MCP](https://github.com/dagger/container-use)** server provides containerized development environments that integrate seamlessly with Goose. This allows you to:
+
+- Work on changes isolated to git branches
+- Run code in containers without affecting your local machine
+- Easily reset and start fresh when needed
+- Maintain clean separation between different projects and experiments
+- Work on things in parallel
+
+## Prerequisites
+
+- Docker installed and running on your system
+- Git installed and configured
+- Goose installed and configured
+
+## Installation
+
+### Step 1: Install Container Use
```

Comment:
The installation steps are still evolving (e.g. `brew` is fresh from this week).

There's 2 options here:

- Either link to our README with the latest instructions
- OR, we make it part of our process to update this README in lockstep with container-use's whenever there are installation changes

Either way is fine by me, do you have a preference?

Reply:
can link to readme - this is so fast moving that is fine